### DISCUSSION TITLE
Concat graph embedding to all node embedding before prediction

### DISCUSSION THREAD

**Discussion by jomalla123** [pyg-team/pytorch_geometric#8273] [Q&A] [answered]
Hi,

Using some GNN layers I have embeddings for each node $h_i$. I would like to concat the graph_embedding $G$ (e.g. mean of all $h_i$) to each $h_i$, before passing this to some linear layers to make a prediction. However, since I working with a batch of graphes with different sizes, I have a hard time figuring out how do that. Is that even possible?

**Comment by rusty1s** [comment, marked as answer]
You can do this via
```python
h = torch.cat([h, h_global[batch]], dim=-1)

**Reply by jomalla123** [reply]
Thanks for the reply, that it very convenient
